### PR TITLE
Android: Enable JNI mode on client too

### PR DESCRIPTION
**Pull request by basilgello**
Signed-off-by: Vasyl Gello <vasek.gello@gmail.com>

### CODE REVIEW

**Comment by rustdesk** [comment]
So it will be called twice?

**Comment by basilgello** [comment]
Service is AFAIK started in isolated process

**Comment by rustdesk** [comment]
Yes, it can be configured to run in isolated process. But we run it in thread. Please do not suggest running it in process, which will introduce a lot of complexity and break the whole RustDesk.

**Comment by basilgello** [comment]
OK :) Then a static singleton should do the trick between MainActivity and MainService in one thread. Actually, `av_jni_set_java_vm` will err out if the same pointer is about to be assigned but we should not rely on it and do things properly.

File: flutter/android/app/src/main/kotlin/com/carriez/flutter_hbb/FFIInitSingleton.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * Static singleton class to initialize NDK contexts in FFI for hardware codecs
+ */
+
+enum class FFIInitSingleton {
+    INSTANCE;
+
+    fun init(ctx: Context) {
+        FFI.init(ctx)
+    }
+}
```
Comment (rustdesk) [inline flutter/android/app/src/main/kotlin/com/carriez/flutter_hbb/FFIInitSingleton.kt:22]: I do not think this is correct, though I have never learned Kotlin.

Comment (rustdesk) [reply flutter/android/app/src/main/kotlin/com/carriez/flutter_hbb/FFIInitSingleton.kt:22]: FFIInitSingleton.INSTANCE is singleton, it does mean init called once.

Comment (rustdesk) [reply flutter/android/app/src/main/kotlin/com/carriez/flutter_hbb/FFIInitSingleton.kt:22]: Chatgpt can help you.
```
import android.content.Context
import java.util.concurrent.atomic.AtomicBoolean

/**
 * Static singleton class to initialize NDK contexts in FFI for hardware codecs
 */

enum class FFIInitSingleton {
    INSTANCE;

    private val initCalled = AtomicBoolean(false)

    fun init(ctx: Context) {
        if (initCalled.compareAndSet(false, true)) {
            FFI.init(ctx)
        } else {
            // Optionally, you can log or handle the case where init has already been called
            println("Init has already been called.")
        }
    }
}
```

Comment (basilgello) [reply flutter/android/app/src/main/kotlin/com/carriez/flutter_hbb/FFIInitSingleton.kt:22]: Yup I noticed :) Thanks

**Pull request closed without merging**